tests/dragonfly/cluster_test.py

```diff
@@ -1063,38 +1071,27 @@ async def generate_config():
         keeping = node.slots[num_outgoing:]
         node.next_slots.extend(keeping)
 
-    # some more time fo seeder
-    await asyncio.sleep(1.0)
-
     seeder.stop()
     await fill_task
-    await asyncio.sleep(1.0)
 
-    # Counter that pushes values to a list
-    async def list_counter(key, client: aioredis.RedisCluster):
-        for i in itertools.count(start=1):
-            await client.lpush(key, i)
+    iterations = 0
+    while True:
+        for node in nodes:
+            states = await node.admin_client.execute_command("DFLYCLUSTER", "SLOT-MIGRATION-STATUS")
+            print(states)
+            if not all(s.endswith("FINISHED") for s in states) and not states == "NO_STATE":
+                break
+        else:
+            break
 
-    # Start ten counters
-    counter_keys = [f"_counter{i}" for i in range(10)]
-    counter_connections = [
-        aioredis.RedisCluster(host="localhost", port=nodes[0].instance.port) for _ in range(10)
-    ]
-    counters = [
-        asyncio.create_task(list_counter(key, conn))
-        for key, conn in zip(counter_keys, counter_connections)
-    ]
+        iterations += 1
+        assert iterations < 100
+
+        await asyncio.sleep(0.1)
 
     # Generate capture, capture ignores counter keys
     capture = await seeder.capture()
 
-    # Finalize slot migration
-    for node in nodes:
-        for sync_id in node.sync_ids:
-            assert "OK" == await node.admin_client.execute_command(
-                "DFLYCLUSTER", "SLOT-MIGRATION-FINALIZE", sync_id
-            )
-
     # Stop counters
     for counter in counters:
         counter.cancel()
```

Comment:
I think we push below the old slot configuration and not the new. also I see the code under # Transfer nodes that updates the node configuration only after the push

Reply:
I see you changed it now.. is something fixed?

Reply:
just wanted to check is it the same

Reply:
True, we push the old config, because generate_config takes the values of `node.slots`. We should first update the node objects and only then push the new config